### PR TITLE
[cp-4891] Modify get_resource_limits and add update_resource_limits command

### PR DESCRIPTION
**Pull request by josephl310**
https://coursera.atlassian.net/browse/CP-4891

### Summary
Added commands to allow partners to update resource limits using coursera_autograder sdk

##### All 3 values provided
Input:
```
coursera_autograder update_resource_limits authoringBranch~nJc-PcxXEeqDJgprEGyUfw dx1Mn q5F9W --grader-cpu 1 --grader-memory-limit 2048 --grader-timeout 1000
```
Output:
```
Updated resource Limits for executor with part id q5F9W in item dx1Mn in course authoringBranch~nJc-PcxXEeqDJgprEGyUfw:
New Reserved CPU (vCPUs): 1.0
New Reserved Memory (MiB): 2048
New Wall Clock Timeout (s): 1000
```

##### Some values provided
Input:
```
coursera_autograder update_resource_limits authoringBranch~nJc-PcxXEeqDJgprEGyUfw dx1Mn q5F9W --grader-memory-limit 3072
```
Output:
```
Updated resource Limits for executor with part id q5F9W in item dx1Mn in course authoringBranch~nJc-PcxXEeqDJgprEGyUfw:
New Reserved CPU (vCPUs): 1.0
New Reserved Memory (MiB): 3072
New Wall Clock Timeout (s): 1000
```

##### Invalid CPU
Input:
```
coursera_autograder update_resource_limits authoringBranch~nJc-PcxXEeqDJgprEGyUfw dx1Mn q5F9W --grader-cpu 0.5 --grader-memory-limit 2048 --grader-timeout 1000
```
Output:
```
ERROR:root:Invalid CPU value. Please choose a value of 1, 2, or 4
```

##### Invalid configuration
Input:
```
coursera_autograder update_resource_limits authoringBranch~nJc-PcxXEeqDJgprEGyUfw dx1Mn q5F9W --grader-cpu 1 --grader-memory-limit 1024
```
Output:
```
ERROR:root:
Unable to update executor resources.
CourseId: authoringBranch~nJc-PcxXEeqDJgprEGyUfw
ItemId: dx1Mn
PartId: q5F9W
Status Code: 400
URL: https://api.coursera.org/api/authoringProgrammingAssignments.v3/?action=updateGraderResourceLimits&id=authoringBranch!~nJc-PcxXEeqDJgprEGyUfw~dx1Mn&partId=q5F9W
Response: {"errorCode":null,"message":"reservedMemory 1024 is not valid. With reservedCpu=1024, Memory must be between 2048 to 8192 incremented by 1024 MB.","details":null}
```

### CODE REVIEW

File: coursera_autograder/commands/get_resource_limits.py [hunk 1]
```diff
@@ -129,7 +126,12 @@ def parser(subparsers):
         'get_resource_limits',
         help='Gets the current resource limits of a programming assignment \
             part (autograder).')
-    parser_resources.set_defaults(func=command_resources)
+    parser_resources.set_defaults(func=command_get_resource_limits)
+
+    # parser_resources = subparsers.add_parser(
```
Comment (sashi-coursera) [inline coursera_autograder/commands/get_resource_limits.py:131]: Is this code in comments needed?

File: coursera_autograder/commands/update_resource_limits.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python
+
+# Copyright 2015 Coursera
```
Comment (sashi-coursera) [inline coursera_autograder/commands/update_resource_limits.py:3]: Can you change the copyright to 2021, at least for the files you touched?

File: coursera_autograder/commands/update_resource_limits.py [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+    params = 'id=%s&partId=%s' % (course_branch_item, args.part)
+    body = {
+        "reservedCpu": int(args.cpu) if args.cpu != None else None, 
+        "reservedMemory": int(args.memory) if args.memory != None else None, 
+        "wallClockTimeout": int(args.timeout) if args.timeout != None else None
+        }
+    result = s.post(args.updateGraderResourceLimits_endpoint, params = params, json = body)
+    if result.status_code == 404:
+        logging.error(
+            '\nUnable to find executor with part id %s in item %s in course %s.\n'
```
Comment (sashi-coursera) [inline coursera_autograder/commands/update_resource_limits.py:50]: How about this? 404, could mean either part doens't exist, or part doesn't have executor id?
```suggestion
            '\nUnable to find the part or executor with part %s in item %s in course %s.\n'
```

**Review by sashi-coursera — APPROVED** [review]
Looks good. One minor (nit) comment.

File: coursera_autograder/commands/get_resource_limits.py (hunk 1 above)
Comment (josephl310) [reply coursera_autograder/commands/get_resource_limits.py:131]: I completely forgot about that. Thanks!

File: coursera_autograder/commands/update_resource_limits.py (hunk 3 above)
Comment (josephl310) [reply coursera_autograder/commands/update_resource_limits.py:50]: Good point. I'll also update the wording from executor to grader to make it consistent with everything else.

**Pull request merged**